akka-persistence-testkit/src/main/scala/akka/persistence/testkit/internal/EventSourcedBehaviorTestKitImpl.scala

```diff
@@ -233,4 +242,16 @@ import akka.stream.scaladsl.Sink
     }
   }
 
+  override def initialize(stateOption: Option[State], events: Event*): Unit = {
+    clear()
+
+    stateOption.foreach(
+      state =>
+        snapshotTestKit.fold(
+          throw new IllegalArgumentException("Cannot initialize from state when snapshots are not used."))(
+          _.persistForRecovery(persistenceId.id, (SnapshotMeta(0), state))))
+    persistenceTestKit.persistForRecovery(persistenceId.id, events)
```

Comment:
Not important and not blocking for merging. I'm just expressing my opinion.

I find that in general the usage of `Option.fold` doesn't help in readability. We need to parse the lines to find  the parenthesis`()()` and it gives very often a bad formatting. 

The following two variants looks more structured and easier to read to me.

```scala
stateOption.foreach { state =>
  snapshotTestKit match {
    case Some(kit) => kit.persistForRecovery(persistenceId.id, (SnapshotMeta(0), state))
    case _ => throw new IllegalArgumentException("Cannot initialize from state when snapshots are not used.")
  }
}
```
or 

```scala
stateOption.foreach { state =>
  snapshotTestKit
    .getOrElse(throw new IllegalArgumentException("Cannot initialize from state when snapshots are not used."))
    .persistForRecovery(persistenceId.id, (SnapshotMeta(0), state))
}
```

But as said, that's just code cosmetics and personal opinion. Feel free to change if you believe it improves. Or to ignore if you don't agree. 😄

Reply:
I do understand the yuckiness that you are expressing, I had a similar feeling where it seems like every approach has some of it 😄 I will think on it more and pick an approach to clean this bit of code up a bit.